### PR TITLE
Add support for filtering asset reports

### PR DESCRIPTION
**Pull request by notthefakestephen**

### CODE REVIEW

File: src/test/java/com/plaid/client/integration/AccountsBalanceGetTest.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public void testAccountsBalanceGetSuccess() throws Exception {
 
     // sandbox should return expected accounts
     List<Account> accounts = response.body().getAccounts();
-    assertEquals(4, accounts.size());
+    assertEquals(5, accounts.size());
```
Comment (notthefakestephen) [inline src/test/java/com/plaid/client/integration/AccountsBalanceGetTest.java]: I believe this change is related to the new account added to the sandbox environment.

File: src/test/java/com/plaid/client/integration/AssetReportFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    assertSuccessResponse(response);
+
+    assertNotNull(respBody.getReport());
+    assertNotNull(respBody.getReport().getUser());
+    assertFalse(respBody.getReport().getItems().isEmpty());
+
+    assertNotNull(respBody.getReport().getAssetReportId());
+
+    List<String> accountIdsToExclude = new ArrayList<>();
+    accountIdsToExclude.add(respBody.getReport().getItems().get(0).getAccounts().get(0).getAccountId());
```
Comment (davidzhanghp) [inline src/test/java/com/plaid/client/integration/AssetReportFilterTest.java]: why exclude the first one? can you add a comment

File: src/test/java/com/plaid/client/integration/AssetReportFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    AssetReportFilterRequest assetReportFilterRequest = new AssetReportFilterRequest(assetReportToken, accountIdsToExclude);
+
+    Response<AssetReportCreateResponse> assetReportFilterResponse = filterAssetReport(assetReportFilterRequest);
+
+    String assetReportFilterToken = assetReportFilterResponse.body().getAssetReportToken();
+
+    // wait till the filtered asset reponse is ready
+    Response<AssetReportGetResponse> filteredReportResponse = waitTillReady(client(), assetReportFilterToken);
+
+    assertEquals(getAssetReportAccountIds(respBody.getReport()).size() - 1,
```
Comment (davidzhanghp) [inline src/test/java/com/plaid/client/integration/AssetReportFilterTest.java]: can you assert on the exact tokens? the size doesn't tell us much

File: src/test/java/com/plaid/client/integration/AssetReportFilterTest.java (hunk 2 above)
Comment (notthefakestephen) [reply src/test/java/com/plaid/client/integration/AssetReportFilterTest.java]: ⚡️

File: src/test/java/com/plaid/client/integration/AssetReportFilterTest.java (hunk 3 above)
Comment (notthefakestephen) [reply src/test/java/com/plaid/client/integration/AssetReportFilterTest.java]: yes.

**Pull request merged**